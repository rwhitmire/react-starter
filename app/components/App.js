import React, { Component } from 'react'
import { fetchJSON } from 'helpers/fetch'
import { makeCancelable } from 'helpers/promise'
import img from './icon.png'

class App extends Component {
  constructor() {
    super()

    this.state = {
      posts: []
    }
  }

  componentDidMount() {
    this.cancelablePromise = makeCancelable(
      fetchJSON('https://www.reddit.com/r/programming.json')
    )

    this.cancelablePromise.promise
      .then(json => {
        const posts = json.data.children.map(item => item.data)
        this.setState({ posts })
      })
      .catch(err => {
        if(err.isCanceled) { console.debug('promise canceled.') }
      })
  }

  componentWillUnmount() {
    this.cancelablePromise.cancel()
  }

  render() {
    return (
      <div>
        <img src={img}/>
        <ul>
          {this.state.posts.map(post => {
            return (
              <li key={post.id}>
                <a href={post.url}>{post.title}</a>
              </li>
            )
          })}
        </ul>
      </div>
    )
  }
}

export default App
