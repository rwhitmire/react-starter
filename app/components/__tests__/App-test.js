import React from 'react'
import ReactTestUtils from 'react-addons-test-utils'
import App from '../App'

test('App should render', () => {
  const renderer = ReactTestUtils.createRenderer()
  renderer.render(<App />)
  const tree = renderer.getRenderOutput()
  expect(tree).toMatchSnapshot()
})
