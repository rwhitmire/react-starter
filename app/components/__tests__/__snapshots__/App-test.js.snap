exports[`test App should render 1`] = `
<div>
  <img
    src="test-file-stub" />
  <ul>
    
  </ul>
</div>
`;
